lib/evm_database_ops.rb

```diff
@@ -117,7 +117,12 @@ def self.restore(db_opts, connect_opts = {})
       else
         connect_opts[:remote_file_name] ||= File.basename(backup_file_name(action))
         backup_folder = action == :dump ? "db_dump" : "db_backup"
-        uri = File.join(connect_opts[:uri], backup_folder, connect_opts[:remote_file_name])
+        #
+        # If the passed in URI contains query parameters, ignore them
+        # when creating the dump file name. They'll be used in the session object.
+        #
+        connect_opts_uri = connect_opts[:uri].split('?')[0]
+        uri = File.join(connect_opts_uri, backup_folder, connect_opts[:remote_file_name])
```

Comment:
This is the only part that I have been touching in all of these changes, but I think you need to tweak what you are editing here (could be wrong, so feel free to correct me):

First, off, this method is going to be changing a bit with https://github.com/ManageIQ/manageiq/pull/17894, but this (hopefully) shouldn't affect you here, if at all.  Just bring that up to avoid more surprises.

But, based on what you have in https://github.com/ManageIQ/manageiq-gems-pending/pull/371, I think you want to be editing `connect_opts[:uri]` instead of `uri`, since that is what gets passed to `query_params`, and not `uri`.  Otherwise you wouldn't be setting the `@region`,`@api_version`, `@domain_id`, and `@security_protocol` at all.

Note:  besides this change, the [`MiqObjectStore` form](https://github.com/ManageIQ/manageiq-gems-pending/compare/master...NickLaMuro:swift-storage-as-object-storage) of the swift changes should not require a single change after #17894 is merged.  That said, I currently have no way of validating the swift code at the moment, which is why I am not going to be opening up a PR with that change.  Just using it as an example of of how to work with the `MiqFileStorage` changes that have recently been merged.

Reply:
@NickLaMuro , not sure what you mean by "editing connect_opts[:uri] instead of uri" - but since I know that I am indeed setting those variables in the Swift Session object, you may not be entirely correct with paragraph 3 above. Am I not indeed editing connect_opts[:uri] in that call to ".split"? Thanks in advance.

Reply:
@jerryk55 Nope, you are right, I got things flipped around in my head.  Clearly I have been looking at this code too much and forgot which way is up.

This looks right, sorry for the confusion.